doc/sphinx/src/fortran.rst

```diff
@@ -0,0 +1,693 @@
+     character(kind=c_char), dimension(*), intent(in) :: filename
+     integer(c_int), value, intent(in) :: id
+     integer(kind=c_int), dimension(:), target, optional, intent(inout) :: sg_mods_enabled
+     real(kind=8), dimension(:), target, optional, intent(inout)        :: sg_mods_values
+   end function init_sg_SpinerDependsRhoSie_f
+
+Initializes a tabulated EOS model that depends on density (ρ) and specific internal energy (sie) using the Spiner interpolation library. Available when built with ``SINGULARITY_USE_SPINER_WITH_HDF5``.
+
+- ``matindex``: Material index (1-based in Fortran)
+- ``eos``: EOS array
+- ``filename``: Path to the HDF5 file containing the tabulated data
+- ``id``: Material ID in the HDF5 file
+- ``sg_mods_enabled``: Optional array for enabling modifiers
+- ``sg_mods_values``: Optional array for modifier values
+
+.. code-block:: fortran
+
+   integer function init_sg_eospac_f(matindex, eos, id, eospac_opts_values, sg_mods_enabled, sg_mods_values) result(err)
+     integer(c_int), value, intent(in) :: matindex
+     type(sg_eos_ary_t), intent(in)    :: eos
+     integer(c_int), value, intent(in) :: id
+     real(kind=8), dimension(:), target, intent(in) :: eospac_opts_values
+     integer(kind=c_int), dimension(:), target, optional, intent(inout) :: sg_mods_enabled
+     real(kind=8), dimension(:), target, optional, intent(inout)        :: sg_mods_values
+   end function init_sg_eospac_f
+
+Initializes an EOSPAC equation of state model at the specified material index. Available when built with ``SINGULARITY_USE_EOSPAC``.
+
+- ``matindex``: Material index (1-based in Fortran)
+- ``eos``: EOS array
+- ``id``: SESAME material ID
+- ``eospac_opts_values``: Array of EOSPAC-specific options:
+
+  - ``eospac_opts_values(1)``: invert_at_setup (0/1)
+  - ``eospac_opts_values(2)``: insert_data value
+  - ``eospac_opts_values(3)``: monotonicity option
+  - ``eospac_opts_values(4)``: apply_smoothing (0/1)
+  - ``eospac_opts_values(5)``: apply_splitting option
+  - ``eospac_opts_values(6)``: linear_interp (0/1)
+- ``sg_mods_enabled``: Optional array for enabling modifiers
+- ``sg_mods_values``: Optional array for modifier values
+
+
+EOS Evaluation Functions
+-----------------------
+
+The following functions are available for evaluating EOS models:
+
+.. code-block:: fortran
+
+   integer function get_sg_EntropyFromDensityInternalEnergy_f(matindex, &
+     eos, rhos, sies, entropies, len, stride, lambda_data) &
+     result(err)
+     integer(c_int), intent(in) :: matindex, len
+     real(kind=8), dimension(:,:,:), intent(in), target:: rhos, sies
+     real(kind=8), dimension(:,:,:), intent(inout), target:: entropies
+     type(sg_eos_ary_t), intent(in)    :: eos
+     integer(c_int), intent(in), optional :: stride
+     real(kind=8), dimension(:,:,:,:), intent(inout), target, optional::lambda_data
+   end function get_sg_EntropyFromDensityInternalEnergy_f
+
+Calculates entropy from density and specific internal energy.
```

Comment:
It may also be worth noting that some equations of state are not thermodynamically complete, and may lack an entropy.

Reply:
This is noted below
```
Some EOS models do not support Entropy. For more details see :ref:`models <Implemented EOS models>`
```